dstep/translator/Record.d

```diff
@@ -34,32 +34,141 @@ void translatePackedAttribute(Output output, Context context, Cursor cursor)
         output.singleLine("align (1):");
 }
 
+struct BitField
+{
+    string type;
+    string field;
+    uint width;
+}
+
+BitField[] translateBitFields(
+    Context context,
+    Cursor[] cursors)
+{
+    void pad(ref BitField[] bitFields, uint totalWidth)
+    {
+        uint padding = 0;
+
+        for (uint size = 8; size <= 64; size *= 2)
+        {
+            if (totalWidth <= size)
+            {
+                padding = size - totalWidth;
+                break;
+            }
+        }
+
+        if (padding != 0)
+            bitFields ~= BitField("uint", "", padding);
+    }
+
+    BitField[] bitFields;
+    uint totalWidth = 0;
+
+    foreach (cursor; cursors)
+    {
+        auto width = cursor.bitFieldWidth();
+
+        if (width == 0) {
+            pad(bitFields, totalWidth);
+            totalWidth = 0;
+        }
+        else {
+            bitFields ~= BitField(
+                translateType(context, cursor).makeString(),
+                cursor.spelling(),
+                width);
+
+            totalWidth += width;
+        }
+    }
+
+    pad(bitFields, totalWidth);
+
+    return bitFields;
+}
+
+void translateBitFields(
+    Output output,
+    Context context,
+    Cursor[] cursors)
+{
+    import std.range;
+
+    auto bitFields = translateBitFields(context, cursors);
+
+    if (bitFields.length == 1)
+    {
+        auto bitField = bitFields.front;
+
+        output.singleLine(
+            "mixin(bitfields!(%s, \"%s\", %s));",
```

Comment:
I prefer to use the backtick string literal, then it's not necessary to escape the quotes.